src/network/networkd-dhcp4.c

```diff
@@ -234,7 +288,7 @@ static int dhcp_remove_router(Link *link, struct in_addr *address) {
         if (!link->network->dhcp_use_routes)
                 return 0;
 
-        r = sd_dhcp_lease_get_router(link->dhcp_lease, &router);
+        r = sd_dhcp_lease_get_router(lease, &router);
         if (IN_SET(r, 0, -ENODATA)) {
                 log_link_debug(link, "DHCP: No gateway received from DHCP server.");
                 return 0;
```

Comment:
Those error messages talk about stuff "received from the DHCP server". It'll be pretty confusing if any of this is emitted when removing a route for the *old* lease. Maybe this should be updated to not log anything (even at debug level), when operating on old lease?

Reply:
OK. Dropped.